zeus/device/cpu/rapl.py

```diff
@@ -0,0 +1,180 @@
+"""RAPL CPUs.
+
+- RAPL (Running Average Power Limit):
+   RAPL is a technology introduced by Intel that allows for power consumption monitoring and control at the processor and memory subsystem level. It provides mechanisms to enforce power limits and manage thermal conditions effectively.
+
+- Power Zone:
+   A power zone in the context of RAPL refers to a logical grouping of components within the CPU or system that share a common power domain. Each power zone can be monitored and controlled independently. Typical power zones include the entire package, specific cores, and memory subsystems.
+
+- Package:
+   The package refers to the physical CPU chip, which may contain multiple cores and integrated components. In RAPL, the package power domain encompasses the power consumption of all the cores and integrated units within the CPU package.
+
+"""
+
+from __future__ import annotations
+
+import os
+import contextlib
+from typing import Sequence
+
+import zeus.device.cpu.common as cpu_common
+from zeus.device.cpu.common import CpuDramMeasurement
+from zeus.device.exception import ZeusBaseCPUError
+from zeus.utils.logging import get_logger
+
+logger = get_logger(name=__name__)
+
+RAPL_DIR = "/sys/class/powercap/intel-rapl"
+
+
+def rapl_is_available() -> bool:
+    """Check if RAPL is available."""
+    if not os.path.exists(RAPL_DIR):
+        logger.info("RAPL is not supported on this CPU.")
+        return False
+    logger.info("RAPL is available.")
+    return True
+
+
+class ZeusRAPLNotSupportedError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL not supported on CPU."""
```

Comment:
I think initially somewhere in the casebase the word `Wrapper` got mistakenly capitalized. Can you search and replace all of them?